SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityJobPoller.java

```diff
@@ -78,11 +82,31 @@ public void runActionOnPoll() {
     for (SingularityTaskId taskId : activeTaskIds) {
       SingularityRequestWithState request = idToRequest.get(taskId.getRequestId());
 
-      if (request == null || !request.getRequest().isScheduled() || taskManager.hasNotifiedOverdue(taskId)) {
+      if (request == null) {
         continue;
       }
 
       final long runtime = start - taskId.getStartedAt();
+
+      if (!request.getRequest().getRequestType().isLongRunning() &&
+          configuration.getTaskExecutionTimeLimitMillis().isPresent() &&
+          runtime >= configuration.getTaskExecutionTimeLimitMillis().get()) {
```

Comment:
Probably want to check against `request. getTaskExecutionTimeLimitMillis().or(configuration.getTaskExecutionTimeLimitMillis())` here so that the individual request can override the execution time limit